### PR TITLE
Refactoring setPipelineState and shaderOptions

### PR DESCRIPTION
**Pull request by ArkadiuszSarwaAMD**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit ddfaa9587f05e330c310b2c887bb0867d2d8ba10
<details><summary>CTS tests (Failed: 0/207358)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36986/69126 (53.5%)</li>
<li>Failed: 0/69126 (0.0%)</li>
<li>Not Supported: 32140/69126 (46.5%)</li><li>Warnings: 0/69126 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37206/69104 (53.8%)</li>
<li>Failed: 0/69104 (0.0%)</li>
<li>Not Supported: 31898/69104 (46.2%)</li><li>Warnings: 0/69104 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 34225/69128 (49.5%)</li>
<li>Failed: 0/69128 (0.0%)</li>
<li>Not Supported: 34903/69128 (50.5%)</li><li>Warnings: 0/69128 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit bc8efcbcf99e1d97d43e117b9bee52441f405e19
<details><summary>CTS tests (Failed: 0/207358)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36986/69126 (53.5%)</li>
<li>Failed: 0/69126 (0.0%)</li>
<li>Not Supported: 32140/69126 (46.5%)</li><li>Warnings: 0/69126 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37206/69104 (53.8%)</li>
<li>Failed: 0/69104 (0.0%)</li>
<li>Not Supported: 31898/69104 (46.2%)</li><li>Warnings: 0/69104 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 34225/69128 (49.5%)</li>
<li>Failed: 0/69128 (0.0%)</li>
<li>Not Supported: 34903/69128 (50.5%)</li><li>Warnings: 0/69128 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/context/llpcPipelineContext.h [hunk 1]
```diff
@@ -245,6 +251,15 @@ class PipelineContext {
...
 #endif
 
+  // Give the color export state to the middle-end, and/or hash it.
+  void setColorExportState(lgc::Pipeline *pipeline, Util::MetroHash64 *hasher) const;
+
+  // Set vertex input descriptions in middle-end Pipeline, and/or hash them.
+  void setVertexInputDescriptions(lgc::Pipeline *pipeline, Util::MetroHash64 *hasher) const;
+
+  // Give the graphics pipeline state to the middle-end, and/or hash it.
+  void setGraphicsStateInPipeline(lgc::Pipeline *pipeline, Util::MetroHash64 *hasher, unsigned stageMask) const;
```
Comment (nhaehnle) [inline llpc/context/llpcPipelineContext.h:261]: These could also be moved into `GraphicsContext`, right?

**Review by nhaehnle — COMMENTED** [review]
Thanks! One note inline.

**Comment by amdvlk-admin** [comment]
db9844e87e29bcbdbc21747b297d649f50182938 Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/llpc/context/llpcGraphicsContext.cpp:321:5: error: ‘PipelineDumper’ has not been declared
     PipelineDumper::updateHashForVertexInputState(
     ^~~~~~~~~~~~~~
[2787/3046] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/context/llpcRayTracingContext.cpp.o
[2788/3046] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/context/llpcContext.cpp.o
[2789/3046] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/lower/llpcSpirvLowerInstMetaRemove.cpp.o

**Comment by amdvlk-admin** [comment]
249c0899abc8ce81e7819d561c12ad40a29d7d61 Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/llpc/context/llpcGraphicsContext.cpp:322:5: error: ‘PipelineDumper’ has not been declared
     PipelineDumper::updateHashForVertexInputState(
     ^~~~~~~~~~~~~~
[2790/3046] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/context/llpcRayTracingContext.cpp.o
[2791/3046] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/context/llpcPipelineContext.cpp.o
[2792/3046] Building CXX object compiler/llpc/CMakeFiles/llpcinternal.dir/lower/llpcSpirvLowerInstMetaRemove.cpp.o

**Comment by amdvlk-admin** [comment]
### Test summary for commit 434f98783da6d3ba1fb6c3b050a17bdfe4dc276c
<details><summary>CTS tests (Failed: 0/208306)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36989/69442 (53.3%)</li>
<li>Failed: 0/69442 (0.0%)</li>
<li>Not Supported: 32453/69442 (46.7%)</li><li>Warnings: 0/69442 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37209/69420 (53.6%)</li>
<li>Failed: 0/69420 (0.0%)</li>
<li>Not Supported: 32211/69420 (46.4%)</li><li>Warnings: 0/69420 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 34225/69444 (49.3%)</li>
<li>Failed: 0/69444 (0.0%)</li>
<li>Not Supported: 35219/69444 (50.7%)</li><li>Warnings: 0/69444 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
Thank you!

**Pull request merged**